### PR TITLE
Fix Mac build, moving linux specific code to its own files

### PR DESCRIPTION
**Pull request by ericvn**
**Please provide a description of this PR:**

Fixes: https://github.com/istio/istio/issues/40087
Fix the issue with the build failing on Macs

### CODE REVIEW

File: tools/istio-iptables/pkg/capture/run_linux.go [hunk 1]
```diff
@@ -84,3 +85,21 @@ func configureTProxyRoutes(cfg *config.Config) error {
 	}
 	return nil
 }
+
+func nsContainerCode(cfg *config.Config) error {
```
Comment (ericvn) [inline tools/istio-iptables/pkg/capture/run_linux.go:89]: Might be able to come up with a better function name.

Comment (ericvn) [reply tools/istio-iptables/pkg/capture/run_linux.go]: Just moved whole function.

**Comment by ericvn** [comment]
I do note that `make recommit` still fails on a Mac, but not in the pipeline, with:
```
...
INFO [runner] linters took 1.947750552s with stages: goanalysis_metalinter: 816.240192ms, structcheck: 72.705µs
cni/pkg/cmd/root.go:88:32: SA4023: this comparison is always true (staticcheck)
		if err = installer.Run(ctx); err != nil {
		                             ^
cni/pkg/install/install.go:80:22: SA4023(related information): (*istio.io/istio/cni/pkg/install.Installer).Run never returns a nil interface value (staticcheck)
func (in *Installer) Run(ctx context.Context) (err error) {
                     ^
INFO File cache stats: 2 entries of total size 21.5KiB
...
```

**Review by howardjohn — APPROVED** [review]

**Comment by GregHanson** [comment]
@ericvn does this need to be cherrypicked to 1.15?

**Comment by ericvn** [comment]
Thanks for the catch @GregHanson

**Pull request merged**